src/arrayMethodJoin.js

```diff
@@ -5,7 +5,31 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    const arrayLength = this.length;
+
+    if (arrayLength === 0) {
+      return '';
+    }
+
+    const finalSeparator = separator === undefined
+      ? ',' : separator === null ? 'null' : separator;
```

Comment:
You could replace this long `if...else` chain with something else... Maybe with a `switch` statement.